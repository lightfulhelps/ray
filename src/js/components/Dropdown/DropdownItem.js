// @flow
import * as React from 'react';
import classNames from 'classnames';
import Icon from '../Icon/Icon';
import type { IconNameType } from '../Icon/icons';

type Props = {
  className?: string,
  icon?: IconNameType,
  label: string,
};

const DropdownItem = ({ className, label, icon, onClick }: Props) => {
  const classes = classNames(className, 'dropdown-item');
  return (
    <a className={classes} href="#" onClick={onClick}>
      <span className="w-100 d-flex align-items-center dropdown-item-content">
        {icon && <Icon name={icon} className="mr-1" />}
        <span>{label}</span>
      </span>
    </a>
  );
};

export default DropdownItem;
