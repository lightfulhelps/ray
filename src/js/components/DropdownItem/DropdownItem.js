// @flow
import * as React from 'react';
import classNames from 'classnames';
import { Icon } from '../../';
import type { IconNameType } from '../Icon/icons';

export type DropdownItemType = {
  icon?: IconNameType,
  iconColor?: string,
  label: string,
  onClick?: () => void,
};

type Props = DropdownItemType & {
  className?: string,
};

const DropdownItem = ({ className, label, icon, iconColor, onClick = () => {} }: Props) => {
  const classes = classNames(className, 'dropdown-item');

  return (
    <div className={classes} onClick={onClick}>
      {icon && <Icon name={icon} color={iconColor} className="mr-1" />}
      <span>{label}</span>
    </div>
  );
};

export default DropdownItem;
