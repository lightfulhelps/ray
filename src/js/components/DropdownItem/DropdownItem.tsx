import * as React from 'react';
import classNames from 'classnames';

export type Props = {
  children: React.ReactNode;
  className?: string;
  isActive?: boolean;
  isDisabled?: boolean;
  isHeader?: boolean;
  onClick?: (arg0: React.MouseEvent) => void;
  tag?: keyof JSX.IntrinsicElements;
  [key: string]: any;
};

const DropdownItem: React.FC<Props> = ({
  children,
  className,
  isActive,
  isDisabled,
  isHeader,
  onClick,
  tag: Tag = 'div',
  ...other
}: Props) => {
  const classes = classNames(
    className,
    { 'dropdown-item': !isHeader },
    { 'dropdown-header': isHeader },
    { active: isActive },
    { disabled: isDisabled },
    { 'cursor-pointer': onClick && !isDisabled }
  );

  function handleClick(e: React.MouseEvent) {
    if (isDisabled) {
      e.preventDefault();
      return;
    }

    if (typeof onClick === 'function') onClick(e);
  }

  return (
    <Tag {...other} className={classes} onClick={handleClick}>
      {children}
    </Tag>
  );
};

export default DropdownItem;
