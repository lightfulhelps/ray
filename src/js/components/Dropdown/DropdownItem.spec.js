import React from 'react';
import merge from 'lodash/merge';
import { shallow } from 'enzyme';
import DropdownItem from './DropdownItem';

const setup = (overrides = {}) => {
  const props = merge(
    {
      icon: 'edit',
      label: 'Edit',
      onClick: jest.fn(),
    },
    overrides
  );
  const wrapper = shallow(<DropdownItem {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<DropdownItem />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render icon if valid icon name is provided', () => {
    const { wrapper } = setup({ icon: null });

    expect(wrapper.find('Icon')).toHaveLength(0);

    wrapper.setProps({ icon: 'addMedia' });

    expect(wrapper.find('Icon')).toHaveLength(1);
  });

  it('should use onClick prop', () => {
    const { wrapper, props } = setup();

    wrapper.simulate('click');

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
