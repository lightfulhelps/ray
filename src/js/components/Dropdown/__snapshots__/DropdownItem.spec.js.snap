// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> should render 1`] = `
<div
  className="dropdown-item clickable"
  onClick={[Function]}
>
  <span>
    some menu item
  </span>
</div>
`;
